conjur/logic/login_logic.py

```diff
@@ -41,10 +43,13 @@ def get_api_key(cls, ssl_verify, credential_data, password, conjurrc):
         elif ssl_verify and credential_data.machine.startswith("https"):
             # Catches the case where a user does not run in insecure mode but the
             # .conjurrc cert_file entry is empty
-            if conjurrc.cert_file == '':
-                certificate_path = True
-            else:
-                certificate_path = conjurrc.cert_file
+            if conjurrc.cert_file == '' and ssl_verify:
+                raise OperationNotCompletedException(cause="The client was initialized without certificate verification, "
+                                    "even though the command was ran with certificate verification enabled. ",
+                                    solution="To continue communicating with the server insecurely, run the command "
+                                    "again with the --insecure flag. Otherwise, reinitialize the client.")
+
```

Comment:
not sure you want to raise the ui message here. and if so, why in the client.py as well?
I would change OperationNotCompletedException to something else.
For my opinion the OperationNotCompletedException should be used in one of two.
1) the exception caused by unfinished operation
2) you want to indicate to the user that this operation is not completed (but actually all exceptions indicate this)

Reply:
@mbenita-Cyberark where would you catch it? I can raise the error and keep raising it through the many layers but where would we catch it and print this specific error. 

My thoughts were the following:
1. OperationNotCompletedException (now changed to `InvalidOperationException`) is a very general error so I didn't want to catch it in the cli.py because I have that UI message I want to print.
2. I also didn't want to catch it in everyone of the controllers because that is not scalable
3. This error is also appropriate to the SDK whose entrypoint is client.py so I want this message to also be seen by those using the SDK.
4. 
What do you think?

Reply:
1. 100%
2,3. why not here? I think the controller is in charge of the UI interaction and therefore it should raise the exception with the UI message.
correct me if I'm wrong the sdk can not use the --insecure flag the way it is written in the error message.
The error for the sdk can be thrown here (with some message changes) as it is now.
4.

Reply:
they can through the SDK. See `ssl_verify`. when conjurrc's cert_file='' and the user feeds ssl_verify=False then the error should be raised (with that message)
See init of client
```
    def __init__(self,
                 account=None,
                 api_key=None,
                 ca_bundle=None,
                 debug=False,
                 http_debug=False,
                 login_id=None,
                 password=None,
                 ssl_verify=True,
                 url=None):
```

Reply:
> I think the controller is in charge of the UI interaction and therefore it should raise the exception with the UI message.

Agree and didnt feel right putting it here